config/identity.go

```diff
@@ -35,25 +35,30 @@ type identityJSON struct {
 	PrivateKey string `json:"private_key"`
 }
 
-// NewIdentity generate a public-private keypair and returns a new Identity.
+// NewIdentity returns a new random identity.
 func NewIdentity() (*Identity, error) {
+	ident := &Identity{}
+	err := ident.Default()
+	return ident, err
+}
+
+// Default generates a random keypair for this identity.
+func (ident *Identity) Default() error {
```

Comment:
I did `Default()` more as a way to keep paralellism with other Configs and because it came handy on a previous code iteration. After trying to take the advice here and revert it (no point in making a change that's not needed), I realize that there is no way to use the ConfigHelper with a valid identity if it could not be loaded from disk otherwise.

Therefore, I like that the identity in the ConfigHelper can be `Default()`'ed. And being key-generation an expensive thing to do, I don't want this to happen everytime a ConfigHelper is created but only on-demand.

So in the end I'm coming to the conclusion I prefer things like they are as it manages both configs and identities with the ConfigHelper a bit more consistently.